credential_provider/pod_identity_credential_provider.go

```diff
@@ -117,94 +55,55 @@ func (p *podIdentityTokenFetcher) FetchToken(ctx credentials.Context) ([]byte, e
 	}
 
 	// Use the K8s API to fetch the token associated with service account
-	tokRsp, err := p.k8sClient.ServiceAccounts(p.nameSpace).CreateToken(ctx, p.svcAcc, &authv1.TokenRequest{
-		Spec: tokenSpec,
-	}, metav1.CreateOptions{})
+	tokRsp, err := p.k8sClient.ServiceAccounts(p.nameSpace).CreateToken(
+		context.Background(),
+		p.svcAcc,
+		&authv1.TokenRequest{Spec: tokenSpec},
+		metav1.CreateOptions{})
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	return []byte(tokRsp.Status.Token), nil
+	return tokRsp.Status.Token, nil
 }
 
-func (p *PodIdentityCredentialProvider) GetAWSConfig() (*aws.Config, error) {
-	// Get token for Pod Identity
-	token, tokenErr := p.fetcher.FetchToken(context.Background())
-	if tokenErr != nil {
-		return nil, fmt.Errorf("failed to fetch token: %+v", tokenErr)
-	}
-
-	var config *aws.Config
-	var configErr error
-	if p.preferredEndpoint == preferenceIPv4 || p.preferredEndpoint == preferenceAuto {
```

Comment:
Previously, this logic essentially did the following

* A string of `""` uses IPv4
* A string of `"ipv4"` uses IPv4
* A string of `"ipv6"` uses IPv6 (there's an unnecessary `|| preferenceAuto` below)
* A string of any other value is an error 

Since this string ultimately comes from the user's SecretProviderClass, we just push that validation up to the `Mount()` call, and error earlier. This has the advantage of showing up in pod events (`kubectl describe pod $FAILED_POD`) in addition to logging an error in the provider logs.